Sources/SwiftSyntax/Identifier.swift

```diff
@@ -0,0 +1,31 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+/// An abstraction for sanitized values on a token.
+public struct Identifier: Equatable, Sendable {
+  /// The sanitized `text` of a token.
+  public let name: String
```

Comment:
I just chatted to @rintaro about this.

What we want is two types:
- An `@_spi(RawSyntax) public struct RawIdentifier` that stores the name of the identifier without keeping the `SyntaxArena` alive. This is intended to be used in performance-critical situations that can guarantee that the `SyntaxArena` stays alive and don’t want to pay the ref-counting overhead for it.
- A `public struct Identifier` that wraps `RawIdentifier` and also keeps the syntax arena alive using `RetainedSyntaxArena`. Identifier should then have a computed property `name: String` that returns a Unicode-normalized version of the identifier. `String` is probably what most clients choose to use but for uses in the compiler, we need to be byte accurate because the compiler considers `U+00E0` (à LATIN SMALL LETTER A WITH GRAVE) to be different than `U+0061 U+0300` (a LATIN SMALL LETTER A followed by ̀ COMBINING GRAVE ACCENT) while Swift’s `String` performs Unicode normalization and considers them the same (`print("\u{e0}" == "\u{61}\u{300}")` prints `true`)

Reply:
Hey @ahoppen thanks for following up on this!

I'm looking in to the above and wanted to clarify whether the `RawIdentifier.name` needs to be a `SyntaxText` or a `RawSyntax`

I've pushed a WIP commit which covers the above and uses a `SyntaxText` which mostly works (except for the test `testRawIdentifier()` which I'll look in to after the conclusion of this conversation)

However on attempting to use `RawSyntax` type I'm getting a bit lost in the code from me being so new to this codebase.

Are you able to:
- clarify if we want `RawIdentifier.name` to be a `SyntaxText` or a `RawSyntax`
- look at my latest WIP commit and provide some input on my attempts so far?

Thanks!

Reply:
`RawIdentifier.name` should be a `SyntaxText`. Sorry if that wasn’t clear from my last comment.